$NetBSD: patch-ab,v 1.7 2006/10/06 22:04:00 rillig Exp $

gcc-2.95 does not know the i386 MMX registers used in the assembler
statements, do disable them.

--- configure.orig	2006-03-20 19:42:25.000000000 +0100
+++ configure	2006-10-07 00:03:06.000000000 +0200
@@ -2053,7 +2053,7 @@ case "$host_os" in
 	solaris*)       lv_solaris=yes;;
 	linux*)         lv_linux=yes;;
 	*sco*)          lv_sco=yes;;
-	freebsd*)	lv_freebsd=yes;;
+	freebsd*|dragonfly*)	lv_freebsd=yes;;
 	netbsd*)	lv_netbsd=yes;;
 	openbsd*)	lv_openbsd=yes;;
 	*darwin*)	lv_darwin=yes;;
@@ -24592,13 +24592,13 @@ echo "$as_me: WARNING: $AIX_COMPILE_INFO
             *)
               LV_THREAD_LIBS=error
 	      lv_save_LIBS="$LIBS"
-	      for thread_lib in "" pthread pthread32 pthreads thread dce; do
+	      for thread_lib in "" -lpthread -lpthread32 -lpthreads -lthread -ldce -pthread; do
 		if test x"$thread_lib" = x; then
 			add_thread_lib=""
 			IN=""
 else
-			add_thread_lib="-l$thread_lib"
-			IN=" in -l$thread_lib"
+			add_thread_lib="$thread_lib"
+			IN=" in $thread_lib"
 		fi
 		if test x"$have_threads" = xposix; then
 			defattr=0
@@ -26430,7 +26430,7 @@ build_subdirs="libvisual"
 VISUAL_PKG_CONFIG_DEP=""
 
 VISUAL_CFLAGS=""
-if test x"$LV_THREAD_CFLAGS" == xgthread2; then
+if test x"$LV_THREAD_CFLAGS" = xgthread2; then
   VISUAL_PKG_CONFIG_DEP="$VISUAL_PKG_CONFIG_DEP gthread-2.0"
 elif test x"$LV_THREAD_CFLAGS" != x; then
   VISUAL_CFLAGS=$LV_THREAD_CFLAGS
@@ -28326,7 +28326,11 @@ _______EOF
         if test x$lv_ix86 = xyes ; then
                 cat >>$outfile <<_______EOF
 
+#if defined(__GNUC__) && __GNUC__ <= 2
+#define VISUAL_ARCH_UNKNOWN
+#else
 #define VISUAL_ARCH_X86
+#endif
 _______EOF
         fi
 
